firebase-firestore/src/main/java/com/google/firebase/firestore/index/IndexEntry.java

```diff
@@ -14,44 +14,53 @@
 
 package com.google.firebase.firestore.index;
 
-/**
- * Represents an index entry saved by the SDK in the local storage. Temporary placeholder, since
- * we'll probably serialize the indexValue right away rather than store it.
- */
-// TODO(indexing)
-public class IndexEntry {
-  private final int indexId;
-  private final byte[] arrayValue;
-  private final byte[] directionalValue;
-  private final String uid;
-  private final String documentName;
-
-  public IndexEntry(
-      int indexId, byte[] arrayValue, byte[] directionalValue, String uid, String documentName) {
-    this.indexId = indexId;
-    this.arrayValue = arrayValue;
-    this.directionalValue = directionalValue;
-    this.uid = uid;
-    this.documentName = documentName;
-  }
+import static com.google.firebase.firestore.util.Util.compareByteArrays;
+import static com.google.firebase.firestore.util.Util.nullSafeCompare;
 
-  public int getIndexId() {
-    return indexId;
-  }
+import androidx.annotation.Nullable;
+import com.google.auto.value.AutoValue;
+import com.google.firebase.firestore.model.DocumentKey;
+import com.google.firebase.firestore.util.Util;
 
-  public byte[] getArrayValue() {
-    return arrayValue;
-  }
+/** Represents an index entry saved by the SDK in its local storage. */
+@AutoValue
+public abstract class IndexEntry implements Comparable<IndexEntry> {
```

Comment:
Made this an AutoValue so that we have hash/equals. I am also planning to keep this class so I removed the TODO that says it is a placeholder.

cc @thebrianchen